openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -761,12 +782,74 @@ def parse_smirnoff_from_source(self, source):
             input_format
             for input_format in self._parameter_io_handlers.keys()
         ]
-        msg = f"Source {source} does not appear to be in a known SMIRNOFF encoding.\n"
+        msg = f"Source {source} could not be read. If this is a file, ensure that the path is correct.\n"
+        msg += "If the file is present, ensure it is in a known SMIRNOFF encoding.\n"
         msg += f"Valid formats are: {valid_formats}\n"
-        msg += f"Parsing vailed with the following error:\n{exception_msg}\n"
+        msg += f"Parsing failed with the following error:\n{exception_msg}\n"
         raise IOError(msg)
 
 
+    def to_string(self, format, discard_cosmetic_attributes=True):
+        """
+        Write this Forcefield and all its associated parameters to a string in a given format which
+        complies with the SMIRNOFF spec.
+
+
+        Parameters
+        ----------
+        format : str
```

Comment:
This looks mostly good, although I'm really not sure about having format as just some string literal... it makes it really unclear as to which values it can take (e.g 'Xml, xml, XML, .xml, .offxml, ..., json, Json, JSON... etc) and doesn't lend itself well to static linting. My personal preference would be to have an `Enum` which contains all of the available formats, and to pass this instead.

e.g: `ff_string = ff.to_string(ForceField.Format.OFFXML, True)` or `ff_string = ff.to_string(ForceField.Format.JSON, True)` if we extend support to other formats in the future.